### PR TITLE
Create Federation Cmdlets

### PR DESCRIPTION
**Pull request by bdwill**
Created functions to:
- Get SDDC Federation members
- Invite members to join Federation

Add .gitignore to project and remove .DS_Store files
Modify CheckVCFVersion function to check if API is at least v3.9

### CODE REVIEW

File: powerVCF.psm1 [hunk 1]
```diff
@@ -2580,6 +2580,123 @@ Export-ModuleMember -Function Get-VCFvROPs
 
 ######### End vRealize Suite Operations ##########
 
+######### Start Federation Management ##########
+
+Function Get-SDDCFederation {
```
Comment (PowerVCF) [inline powerVCF.psm1]: All other cmdlets have VCF in the name to distinguish them from non VCF cmdlets.

File: powerVCF.psm1 [hunk 2]
```diff
@@ -2580,6 +2580,123 @@ Export-ModuleMember -Function Get-VCFvROPs
...
+<#
+    .SYNOPSIS
+    Get information on existing Federation
+
+    .DESCRIPTION
+    Gets the complete information about the existing SDDC Federation.
+
+    .EXAMPLE
+    PS C:\> Get-Federation
```
Comment (PowerVCF) [inline powerVCF.psm1]: Example doesnt match the function name

File: powerVCF.psm1 [hunk 3]
```diff
@@ -2580,6 +2580,123 @@ Export-ModuleMember -Function Get-VCFvROPs
...
+Function New-SDDCFederationInvite {
+<#
+    .SYNOPSIS
+    Invite new member to SDDC Federation
+
+    .DESCRIPTION
+    A function that creates a new invitation for a member to join the existing SDDC Federation.
+
+    .EXAMPLE
+    PS C:\> New-SDDCFederationInvite -inviteeFqdn sddc-manager1.vsphere.local
```
Comment (PowerVCF) [inline powerVCF.psm1]: IMHO -fqdn would be enough to keep it simple but your function - your call :)

File: powerVCF.psm1 [hunk 4]
```diff
@@ -2580,6 +2580,123 @@ Export-ModuleMember -Function Get-VCFvROPs
...
+Function Get-SDDCFederationMembers {
+<#
+    .SYNOPSIS
+    A function that gets information on all members in the SDDC Federation
+
+    .DESCRIPTION
+    Gets the complete information about the existing SDDC Federation members.
+
+    .EXAMPLE
+    PS C:\> Get-FederationMembers
```
Comment (PowerVCF) [inline powerVCF.psm1]: Example does not match the function name

File: powerVCF.psm1 (hunk 1 above)
Comment (PowerVCF) [reply powerVCF.psm1]: Same for the other functions below

**Comment by bdwill** [comment]
@PowerVCF thanks for checking. Pulling back PR as I'm working in a new branch now and will submit new PR once new functions are complete.

**Pull request closed without merging**